msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerAccountManagerStrategy.java

```diff
@@ -56,397 +48,305 @@
-                Logger.error(
-                        TAG + methodName,
-                        ErrorStrings.BROKER_REQUEST_CANCELLED,
-                        errorMsg,
-                        e
-                );
-
-                Telemetry.emit(
-                        new BrokerEndEvent()
-                                .putAction(methodName)
-                                .isSuccessful(false)
-                                .putErrorCode(ErrorStrings.BROKER_REQUEST_CANCELLED)
-                                .putErrorDescription(errorMsg)
-                );
-
-                throw new ClientException(
-                        ErrorStrings.BROKER_REQUEST_CANCELLED,
-                        errorMsg,
-                        e
-                );
-            } catch (final AuthenticatorException e) {
-                final String errorMsg = "AuthenticatorException thrown when talking to account manager. The broker request cancelled.";
-                Logger.error(
-                        TAG + methodName,
-                        ErrorStrings.BROKER_REQUEST_CANCELLED,
-                        errorMsg,
-                        e
-                );
-
-                Telemetry.emit(
-                        new BrokerEndEvent()
-                                .putAction(methodName)
-                                .isSuccessful(false)
-                                .putErrorCode(ErrorStrings.BROKER_REQUEST_CANCELLED)
-                                .putErrorDescription(errorMsg)
-                );
-
-                throw new ClientException(
-                        ErrorStrings.BROKER_REQUEST_CANCELLED,
-                        errorMsg,
-                        e
-                );
-            } catch (final IOException e) {
-                final String errorMsg = "IOException thrown when talking to account manager. The broker request cancelled.";
-                // Authenticator gets problem from webrequest or file read/write
-                Logger.error(
-                        TAG + methodName,
-                        ErrorStrings.BROKER_REQUEST_CANCELLED,
-                        errorMsg,
-                        e
-                );
-
-                Telemetry.emit(
-                        new BrokerEndEvent()
-                                .putAction(methodName)
-                                .isSuccessful(false)
-                                .putErrorCode(ErrorStrings.BROKER_REQUEST_CANCELLED)
-                                .putErrorDescription(errorMsg)
-                );
-
-                throw new ClientException(
-                        ErrorStrings.BROKER_REQUEST_CANCELLED,
-                        errorMsg,
-                        e
-                );
-            }
+        return result;
+    }
+
+    private static BaseException getBaseExceptionFromException(final Exception e) {
```

Comment:
We already have a method like this in `ExceptionAdapter.java` within common, why not use that?

Reply:
It's not exactly same. (i.e. this returns an error specific to this class if the exception can't be matched).

Reply:
May be we can add that as a case to the method in common? 

Leaving it here is fine too. I'll let you pick.

Reply:
We can't. This generic error message is specific to account manager.